WooCommerce/Classes/System/WooNavigationController.swift

```diff
@@ -63,27 +62,6 @@ private class WooNavigationControllerDelegate: NSObject, UINavigationControllerD
     func navigationControllerPreferredInterfaceOrientationForPresentation(_ navigationController: UINavigationController) -> UIInterfaceOrientation {
         forwardDelegate?.navigationControllerPreferredInterfaceOrientationForPresentation?(navigationController) ?? .portrait
     }
-
-    /// Forwards the event to the children delegate.
-    ///
-    func navigationController(
-        _ navigationController: UINavigationController,
-        interactionControllerFor animationController: UIViewControllerAnimatedTransitioning
-    ) -> UIViewControllerInteractiveTransitioning? {
-        forwardDelegate?.navigationController?(navigationController, interactionControllerFor: animationController)
-    }
-
-    /// Forwards the event to the children delegate.
-    ///
-    func navigationController(_ navigationController: UINavigationController,
-                              animationControllerFor operation: UINavigationController.Operation,
-                              from fromVC: UIViewController,
-                              to toVC: UIViewController) -> UIViewControllerAnimatedTransitioning? {
-        forwardDelegate?.navigationController?(navigationController,
-                                               animationControllerFor: operation,
-                                               from: fromVC,
-                                               to: toVC)
-    }
```

Comment:
It seems that this code was added here https://github.com/woocommerce/woocommerce-ios/pull/3872 by Ernesto, but I'm not sure it is currently used.
@Ecarrion as far as you remember, removing this code, does something unexpected happen in other parts of the app?

Reply:
Interesting problem!

I totally missed that providing an empty implementation for these methods breaks the interactive pop gesture. 🤦 

These methods are implemented and forwarded to allow subclasses of `WooNavigationController.swift` to respond to `UINavigationControllerDelegate` as needed.

By deleting these two methods we won't allow subclasses to provide a custom animation/interaction, which is something we don't need at the moment, thankfully.

As solving this problem is not trivial, I think the quickest win is to delete these two methods and to add a comment on why they are not present anymore. As well as linking/creating a GH issue for this problem.

What do you think @pmusolino @itsmeichigo ?

Reply:
@Ecarrion So sorry for not explicitly having mentioned you in this issue earlier 🙇‍♀️ I actually believe that since the implementation was not intended, it's safe to just remove it for now. In the future if we wish to customize transition animations, it'll be part of the task and not what we have to worry about for now. Does that make sense?

Reply:
That makes sense! But I still think we should add a comment explaining why those methods are missing and probably best to create an issue with the `tech debt` label so we can monitor and remember it.

Reply:
@Ecarrion I honestly don't think this is a tech debt because we don't need that for now. I mean those delegate methods are optional, so the person who takes care of the transition animation in the future should be able to add them in as part of the task.

Reply:
It makes sense to me to remove the code and about the comment, it costs us nothing to do it. I'm not sure about the issue, since, like said by @itsmeichigo, it's not a tech debt and I don't think we should track it.

Reply:
FYI I just added the extra comment in c753050 🙂

Reply:
I say it's tech debt because let's imagine the following scenario:

- We need to create a new navigation controller (Subclass of `WooNavigationController`) with some custom interactions.

- Dev will create the subclass and will implement those two methods to provide a custom interaction.

- It will not work, because the methods are not being forwarded from `WooNavigationController`

- Dev will add and forward those methods in `WooNavigationController` but that will make all other subclasses like `WooTabNavigationController` to break and lose it's native pop gesture interaction.

- Dev will realize that our current solution does not scale and will have to find alternatives.

To me, that sounds like tech debt 😅. And creating an issue explaining the situation will probably save a day or two for any future team member. 🪙🪙

Reply:
Thanks @Ecarrion for the clarification 💯 This is really something to be aware of - I have created the issue here: #4524.